core/ts/MaciState.ts

```diff
@@ -179,8 +180,10 @@ export class MaciState implements IMaciState {
       maciState.stateTree.insert(leafHash);
     }
 
-    // re-generate the polls and set the maci state ref
-    maciState.polls = json.polls.map((jsonPoll: IJsonPoll) => Poll.fromJSON(jsonPoll, maciState));
+    // re-generate the polls and set the maci state reference
+    maciState.polls = new Map(
+      json.polls.map((jsonPoll: IJsonPoll, index) => [BigInt(index), Poll.fromJSON(jsonPoll, maciState)]),
+    );
```

Comment:
Let's add new line before return.
```suggestion
    // re-generate the polls and set the maci state reference
    maciState.polls = new Map(
      json.polls.map((jsonPoll: IJsonPoll, index) => [BigInt(index), Poll.fromJSON(jsonPoll, maciState)]),
    );

```